File: Data Structures & Algorithms/two-integer-sum/submission-17.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_table = {}

        for i, n in enumerate(nums):
            complement = target - n
            if complement in hash_table:
                return [hash_table[complement], i]
            hash_table[n] = i
            

File: Data Structures & Algorithms/binary-search/submission-17.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        let lo = 0, hi = nums.length - 1;
        while (lo<=hi){
            let m = Math.floor((lo + hi) / 2);
            if (nums[m] === target)   return m;
            else if (nums[m] > target)    hi = m -1;
            else    lo = m + 1;
        }
        return -1;
    }
}
